File: Wff.Desktop/Converters.fs
namespace Wff.Converters

open Avalonia.Data.Converters

type StringEqualConverter() =
    interface IValueConverter with
        member this.Convert
            (
                value: obj | null,
                targetType: System.Type,
                parameter: obj | null,
                culture: System.Globalization.CultureInfo
            ) : obj | null =
            match value, parameter with
            | (:? string as sval), (:? string as param) -> box (sval = param)
            | _ -> box false

        member self.ConvertBack
            (
                value: obj | null,
                targetType: System.Type,
                parameter: obj | null,
                culture: System.Globalization.CultureInfo
            ) : obj | null =
            raise (System.NotImplementedException())


type StringNotEqualConverter() =
    interface IValueConverter with
        member this.Convert
            (
                value: obj | null,
                targetType: System.Type,
                parameter: obj | null,
                culture: System.Globalization.CultureInfo
            ) : obj | null =
            match value, parameter with
            | (:? string as sval), (:? string as param) -> box (sval <> param)
            | _ -> box false

        member self.ConvertBack
            (
                value: obj | null,
                targetType: System.Type,
                parameter: obj | null,
                culture: System.Globalization.CultureInfo
            ) : obj | null =
            raise (System.NotImplementedException())


open System
open System.Globalization
open Avalonia.Data.Converters
open System.Collections.Generic

type RegionContentConverter() =
    interface IMultiValueConverter with
        member _.Convert
            (values: IList<obj | null>, targetType: Type, parameter: obj | null, culture: CultureInfo)
            : obj | null =

            let region =
                match values.[0] with
                | :? string as v -> v
                | _ -> ""

            let notEqual =
                match values.[1] with
                | :? bool as b -> b
                | _ -> false

            if notEqual then
                box (sprintf "Select Region %s" region)
            else
                box "Select Region"

File: Wff.Desktop/Converters.cs
namespace Wff.Converters;
using System;
using System.Collections.Generic;
using System.Globalization;
using Avalonia.Data.Converters;

public class StringEqualConverter : IValueConverter
{
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        if (value is string sval && parameter is string param)
        {
            return sval == param;
        }
        return false;
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}

public class StringNotEqualConverter : IValueConverter
{
    public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        if (value is string sval && parameter is string param)
        {
            return sval != param;
        }
        return false;
    }

    public object? ConvertBack(object? value, Type targetType, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}

public class RegionContentConverter : IMultiValueConverter
{
    public object? Convert(IList<object?> values, Type targetType, object? parameter, CultureInfo culture)
    {
        string region = values[0] is string v ? v : "";
        bool notEqual = values[1] is bool b ? b : false;

        if (notEqual)
        {
            return $"Select Region {region}";
        }
        else
        {
            return "Select Region";
        }
    }

    public object?[] ConvertBack(object? value, Type[] targetTypes, object? parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
